cmd/nerdctl/main.go

```diff
@@ -52,152 +48,124 @@ func xmain() error {
-	app.EnableBashCompletion = true
-	app.Version = strings.TrimPrefix(version.Version, "v")
-	app.Flags = []cli.Flag{
-		&cli.BoolFlag{
-			Name:  "debug",
-			Usage: "debug mode",
-		},
-		&cli.BoolFlag{
-			Name:  "debug-full",
-			Usage: "debug mode (with full output)",
-		},
-		&cli.StringFlag{
-			Name:    "address",
-			Aliases: []string{"a", "host", "H"},
-			Usage:   "containerd address, optionally with \"unix://\" prefix",
-			EnvVars: []string{"CONTAINERD_ADDRESS"},
-			Value:   defaults.DefaultAddress, // same for rootless as well (mount-namespaced)
-		},
-		&cli.StringFlag{
-			Name:    "namespace",
-			Aliases: []string{"n"},
-			Usage:   "containerd namespace, such as \"moby\" for Docker, \"k8s.io\" for Kubernetes",
-			EnvVars: []string{namespaces.NamespaceEnvVar},
-			Value:   namespaces.Default,
-		},
-		&cli.StringFlag{
-			Name:    "snapshotter",
-			Aliases: []string{"storage-driver"},
-			Usage:   "containerd snapshotter",
-			EnvVars: []string{"CONTAINERD_SNAPSHOTTER"},
-			// TODO: change the default to "native" when overlayfs is unavailable (on rootless)
-			Value: containerd.DefaultSnapshotter,
-		},
-		&cli.StringFlag{
-			Name:  "cni-path",
-			Usage: "Set the cni-plugins binary directory",
-			// CNI_PATH is from https://www.cni.dev/docs/cnitool/
-			EnvVars: []string{"CNI_PATH"},
-			Value:   ncdefaults.CNIPath(),
-		},
-		&cli.StringFlag{
-			Name:  "cni-netconfpath",
-			Usage: "Set the CNI config directory",
-			// NETCONFPATH is from https://www.cni.dev/docs/cnitool/
-			EnvVars: []string{"NETCONFPATH"},
-			Value:   ncdefaults.CNINetConfPath(),
-		},
-		&cli.StringFlag{
-			Name:  "data-root",
-			Usage: "Root directory of persistent nerdctl state (managed by nerdctl, not by containerd)",
-			Value: ncdefaults.DataRoot(),
-		},
-		// cgroup-manager flag is from Podman.
-		// Because Docker's equivalent is complicated: --exec-opt native.cgroupdriver=(cgroupfs|systemd)
-		&cli.StringFlag{
-			Name:  "cgroup-manager",
-			Usage: "Cgroup manager to use (\"cgroupfs\"|\"systemd\")",
-			Value: ncdefaults.CgroupManager(),
-		},
-		&cli.BoolFlag{
-			Name:  "insecure-registry",
-			Usage: "skips verifying HTTPS certs, and allows falling back to plain HTTP",
-		},
+func newApp() *cobra.Command {
+	var rootCmd = &cobra.Command{
+		Use:     "nerdctl",
+		Short:   "nerdctl is a command line interface for containerd",
+		Version: strings.TrimPrefix(version.Version, "v"),
 	}
-	app.Before = func(clicontext *cli.Context) error {
-		if clicontext.Bool("debug-full") {
-			clicontext.Set("debug", "true")
+	rootCmd.PersistentFlags().Bool("debug", false, "debug mode")
+	rootCmd.PersistentFlags().Bool("debug-full", false, "debug mode (with full output)")
+	rootCmd.PersistentFlags().StringP("address", "H", defaults.DefaultAddress, `containerd address, optionally with "unix://" prefix`)
+	rootCmd.PersistentFlags().String("namespace", namespaces.Default, `containerd namespace, such as "moby" for Docker, "k8s.io" for Kubernetes`)
```

Comment:
`-n` alias seems lost

Reply:
(You can temporarily fork cobra)

Reply:
> (You can temporarily fork cobra)

Yes, I did it, and I create PR for this feature https://github.com/spf13/pflag/pull/325 , https://github.com/spf13/cobra/pull/1487